commerce/domain/tax/cart/calculator/classes/TaxCartCalculatorSample.cls

```diff
@@ -242,9 +249,11 @@ public class TaxCartCalculatorSample extends CartExtension.TaxCartCalculator {
     CartExtension.cartItemPriceAdjustmentList cipaList,
     String id
   ) {
-    for (Integer i = (cipaList.size() - 1); i >= 0; i--) {
-      if (String.valueOf(cipaList.get(i).getId()) == id)
-        return cipaList.get(i);
+    Iterator<CartExtension.CartItemPriceAdjustment> cipaIterator = cipaList.iterator();
+    while (cipaIterator.hasNext()) {
+      CartExtension.CartItemPriceAdjustment cipa = cipaIterator.next();
+      if (String.valueOf(cipa.getId()) == id)
```

Comment:
Please make sure comparison by ids is exactly what you want. Please see https://salesforce.quip.com/8X5fAhaNEODJ#temp:C:TFaf8329189b138478996479b067

Reply:
== will work here because:
1. we are not comparing object just strings so == checks values of strings not address
2. existing logic also using ==
and sanity test is done by comparing results between before these changes and after these changes.

Reply:
I just want to bring to your attention the fact that `cipa.getId()` returns `null` for `CartItemPriceAdjustment` that has been created in the scope of given request processing. It is very hard for me to follow what this apex class is doing really as I am not tax expert. But this code looks risky. I also see that this test class has no any unit test, so it is on you to take a risk or to implement unit tests to prove if this code works as expected.